sympy/ntheory/primetest.py

```diff
@@ -110,6 +110,24 @@ def mr(n, bases):
     return True
 
 
+def isprimeE(n, b):
+    """isprimeE(n ,b) - Test whether n is prime or an Euler pseudoprime to base b."""
+	if (not mr(n, b)): return False
+    
+	r = n-1
+	while (r % 2 == 0): r //= 2
+    
+	c = pow(b, r, n)
+	if (c == 1): return True
+	while (1):
+		if (c == 1): return False
```

Comment:
This test for `c == 1` can appear after recomputing the c since on the first pass we know it can't be true.

Reply:
nice observation sir!
will fix it for sure